Tests/Unit/MSFT_xFirewall.Tests.ps1

```diff
@@ -191,16 +191,176 @@ InModuleScope $DSCResourceName {
+
+                Assert-MockCalled Set-NetFirewallRule -Exactly 1
+                Assert-MockCalled Test-RuleProperties -Exactly 1
+            }
+        }
+        Context 'Ensure is Present and the Firewall Does Exist but has different DisplayGroup' {
+            It "should throw a CantChangeDisplayGroupError exception on rule $($rule.Name)" {
+                Mock Set-NetFirewallRule
+                Mock Test-RuleProperties {return $false}
+
+                $errorId = 'CantChangeDisplayGroupError'
+                $errorCategory = [System.Management.Automation.ErrorCategory]::InvalidOperation
+                $errorMessage = $($LocalizedData.CantChangeDisplayGroupError) -f $Name
+                $exception = New-Object -TypeName System.InvalidOperationException `
+                    -ArgumentList $errorMessage
+                $errorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord `
+                    -ArgumentList $exception, $errorId, $errorCategory, $null
+
+                { $result = Set-TargetResource `
+                    -Name $rule.Name `
+                    -DisplayGroup 'Different' `
+                    -Ensure 'Present' } | Should Throw $errorRecord
+
+                Assert-MockCalled Set-NetFirewallRule -Exactly 0
+                Assert-MockCalled Test-RuleProperties -Exactly 1
+            }
+        }
+        Context 'Ensure is Present and the Firewall Does Exist but has different Enabled' {
+            It "should call expected mocks on firewall rule $($rule.Name)" {
+                Mock Set-NetFirewallRule
+                Mock Test-RuleProperties {return $false}
+                $result = Set-TargetResource `
+                    -Name $rule.Name `
+                    -Enabled 'False' `
+                    -Ensure 'Present'
+
+                Assert-MockCalled Set-NetFirewallRule -Exactly 1
+                Assert-MockCalled Test-RuleProperties -Exactly 1
+            }
+        }
+        Context 'Ensure is Present and the Firewall Does Exist but has different Action' {
+            It "should call expected mocks on firewall rule $($rule.Name)" {
+                Mock Set-NetFirewallRule
+                Mock Test-RuleProperties {return $false}
+                $result = Set-TargetResource `
+                    -Name $rule.Name `
+                    -Action 'Allow' `
+                    -Ensure 'Present'
+
+                Assert-MockCalled Set-NetFirewallRule -Exactly 1
+                Assert-MockCalled Test-RuleProperties -Exactly 1
+            }
+        }
+        Context 'Ensure is Present and the Firewall Does Exist but has different Profile' {
+            It "should call expected mocks on firewall rule $($rule.Name)" {
+                Mock Set-NetFirewallRule
+                Mock Test-RuleProperties {return $false}
+                if ($rule.Profile -ccontains 'Domain') {$NewProfile = @('Public','Private')} else {$NewProfile = @('Domain','Public')}
```

Comment:
Can we put the { on a new line to get it under the character limit?  Also be cool if we put else on it's on line too..

My personal perference:

```powershell
if (test this)
{
   code
}
else
{
   else code
}

Not sure what the style guide is but let's at least break this up a little.